src/serial.rs

```diff
@@ -1862,12 +1862,22 @@ halUsart! {
     UART10: (uart10, apb2enr, 7, uart10en, pclk2),
 }
 
+impl<USART, PINS> fmt::Write for Serial<USART, PINS>
+where
+    Serial<USART, PINS>: serial::Write<u8>,
+{
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        let _ = s.as_bytes().iter().try_for_each(|c| block!(self.write(*c)));
+        Ok(())
```

Comment:
Now I'm afraid it doesn't work, but I suppose it's better if it does:

```suggestion
        ls.as_bytes().iter().try_for_each(|c| block!(self.write(*c)))
```

Now, that's not really critical ;-)

Reply:
I don't think I follow. What do you mean?

Reply:
I thought that the error was Infaillible,  but it's not. So the correct implementation is:

```suggestion
        s.as_bytes().iter().try_for_each(|c| block!(self.write(*c))).map_err(|_| fmt::Error)
```

Else, the serial error will be just silently skipped.

Reply:
I am aware. But completely revamping the implementation was not quite the plan...

Reply:
yeah, that's why I said it was not really critical, out of subject would be more appropriate.